src/network/networkd-json.c

```diff
@@ -873,6 +873,27 @@ static int sip_build_json(Link *link, JsonVariant **ret) {
         return r;
 }
 
+static int captive_portal_build_json(Link *link, JsonVariant **ret) {
+        int r;
+        const char *captive_portal = NULL;
+
+        assert(link);
+        assert(ret);
+
+        if (!link->network || !link->dhcp_lease) {
+                *ret = NULL;
+                return 0;
+        }
+
+        if (link->network->dhcp_use_captive_portal) {
+                r = sd_dhcp_lease_get_captive_portal(link->dhcp_lease, &captive_portal);
+                if (r < 0 && r != -ENODATA)
+                        return r;
```

Comment:
Missing `r == -ENODATA` case handling?

Reply:
Well here I don't really care about the ENODATA case I think. These functions return ENODATA to match the similar getters. In the ENODATA case captive_portal will remain NULL and the dhcp6 and ipv6ra blocks will have a chance to fill it.